File: codes/python/739.每日温度.py
class Solution:
	def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
		stack = []
		res = [0] * len(temperatures)
		for i in range(len(temperatures)):
			while stack and temperatures[stack[-1]] < temperatures[i]:
				res[stack[-1]] = i - stack[-1]
				stack.pop()
			stack.append(i)
		return res

File: codes/java/739.每日温度.java
class Solution {
	public int[] dailyTemperatures(int[] temperatures) {
		int[] res = new int[temperatures.length];
		Stack<Integer> stack = new Stack<>();
		for (int i = 0; i < temperatures.length; i++) {
			while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
				int preIndex = stack.pop();
				res[preIndex] = i - preIndex;
			}
			stack.push(i);
		}
		return res;
	}
}